### PR TITLE
fix(environments): Fix release stats sidebar

### PR DESCRIPTION
**Pull request by lynnagara**
Fix release stats sidebar and ensure release chart triggers rerender

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/components/group/releaseChart.jsx [hunk 1]
```diff
@@ -25,20 +25,27 @@ const GroupReleaseChart = createReactClass({
     title: PropTypes.string,
   },
 
-  getInitialState(props) {
-    if (!defined(props)) props = this.props;
+  getInitialState() {
+    return this.getNextState(this.props);
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/group/releaseChart.jsx]: https://github.com/reactjs/rfcs/blob/master/text/0006-static-lifecycle-methods.md#state-derived-from-propsstate

looks like exactly what getDerivedStateFromProps is trying to solve : O

File: src/sentry/static/sentry/app/components/group/releaseChart.jsx [hunk 2]
```diff
@@ -25,20 +25,27 @@ const GroupReleaseChart = createReactClass({
...
+  componentWillReceiveProps(nextProps) {
+    this.setState(this.getNextState(nextProps));
+  },
+
+  getNextState(props) {
     let releaseStats = props.releaseStats;
     let releasePoints = {};
-    if (defined(releaseStats)) {
-      releaseStats[this.props.statsPeriod].forEach(point => {
+    if (releaseStats) {
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/group/releaseChart.jsx]: what were the defined falsy values that were causing problems here?

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/group/releaseChart.jsx]: There wasn't a problem here, just trying to simplify it

File: src/sentry/static/sentry/app/components/group/releaseChart.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/group/releaseChart.jsx]: Yeah! It's only in 16.3 though

**Review by MaxBittker — APPROVED** [review]
looks good to me 📊

**Pull request merged**